### PR TITLE
Initial PR for moving subnet creation out of Azure Common

### PR DESCRIPTION
**Pull request by jmspring**
PR still needs to merge in multi-cluster / WAF

### CODE REVIEW

**Comment by jmspring** [comment]
This issue addresses -- https://github.com/microsoft/bedrock/issues/1110

File: cluster/azure/subnet/outputs.tf [hunk 1]
```diff
@@ -1,4 +1,4 @@
-output "subnet_id" {
-  description = "The id of the vNet"
-  value       = azurerm_subnet.subnet.id
+output "subnet_ids" {
+  description = "The id of the subnet"
+  value       = azurerm_subnet.subnet.*.id
```
Comment (sarath-p) [inline cluster/azure/subnet/outputs.tf]: Is it outputting all subnet id's that are created?

Comment (jmspring) [reply cluster/azure/subnet/outputs.tf]: There isn't a need, as the subnet is created with the cluster.  Data source uses vnet name and subnet name.  What scenario are you thinking?

Comment (sarath-p) [reply cluster/azure/subnet/outputs.tf]: Got it. I can't think of a scenario now, but curious whether it is writing out `n` number of subnets or the last/first subnet id.

Comment (jmspring) [reply cluster/azure/subnet/outputs.tf]: Also - it's outputting all the subnets created (it takes in lists, so may create multiple).

**Pull request merged**